### PR TITLE
SMD-MDB-CA.CS collStats model soundness

### PR DESCRIPTION
**Pull request by mikhailZakharevich**
added an option for `storageStats` field

### CODE REVIEW

File: src/main/scala/group/scala/karazin/mongo4s/model.scala [hunk 1]
```diff
@@ -192,10 +192,11 @@ object model:
 
     object CollStats:
       final case class LatencyStats(histograms: Boolean)
+      final case class StorageStats(scale: Int)
```
Comment (VLADISLAV008) [inline src/main/scala/group/scala/karazin/mongo4s/model.scala:195]: MongoDB also supports non-integer scale factor (e.g. float). But I'm not sure if this is critical and whether it needs to be fixed.

Comment (mikhailZakharevich) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala:195]: Yes, it does, but it only takes the integer part of a number anyway. Good point though.

**Review by VLADISLAV008 — APPROVED** [review]

**Review by IgorWolkov — APPROVED** [review]

File: src/main/scala/group/scala/karazin/mongo4s/model.scala [hunk 2]
```diff
@@ -192,10 +192,11 @@ object model:
...
 
       // Should be rewritten with a builder
       final case class Command(latencyStats: Option[LatencyStats],
-                               storageStats: Option[EmptyObject],
+                               storageStats: Option[EmptyObject | StorageStats],
```
Comment (IgorWolkov) [inline src/main/scala/group/scala/karazin/mongo4s/model.scala:199]: Could storage stats be `EmptyObject`?

**Review by IgorWolkov — CHANGES_REQUESTED** [review]
Could storage stats be EmptyObject?

File: src/main/scala/group/scala/karazin/mongo4s/model.scala (hunk 2 above)
Comment (mikhailZakharevich) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala:199]: according to the docs:
- Specify an empty document (i.e. storageStats: {}) 
- Specify the scale factor (i.e. storageStats: { scale: <number> })
Either of those

Comment (IgorWolkov) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala:199]: But do we need `Option` then?

Comment (mikhailZakharevich) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala:199]: In the same docs:
The $collStats stage accepts an argument document with the following **optional** fields:...(includes `storageStats`)
Which is kind of confusing

Comment (IgorWolkov) [reply src/main/scala/group/scala/karazin/mongo4s/model.scala:199]: Cool, thanks.

**Review by IgorWolkov — APPROVED** [review]

**Pull request merged**